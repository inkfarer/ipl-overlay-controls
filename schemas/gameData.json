{
    "$schema": "http://json-schema.org/draft-07/schema",
    "type": "array",
    "maxItems": 7,
    "items": {
        "type": "object",
        "properties": {
            "winner": {
                "type": "string",
                "enum": ["none", "alpha", "bravo"]
            },
            "color": {
                "type": "object",
                "allOf": [
                    {
                        "$ref": "defs/color.json#/color"
                    },
                    {
                        "properties": {
                            "colorsSwapped": {
                                "type": "boolean"
                            }
                        },
                        "required": ["colorsSwapped"]
                    }
                ]
            }
        },
        "required": ["winner"]
    },
    "default": [
        {
            "winner": "none",
            "color": {
                "index": 0,
                "title": "Green vs Grape",
                "clrA": "#37FC00",
                "clrB": "#7D28FC",
                "categoryName": "Ranked Modes",
                "colorsSwapped": false
            }
        },
        {
            "winner": "none",
            "color": {
                "index": 1,
                "title": "Green vs Magenta",
                "clrA": "#04D976",
                "clrB": "#D600AB",
                "categoryName": "Ranked Modes",
                "colorsSwapped": false
            }
        },
        {
            "winner": "none",
            "color": {
                "index": 2,
                "title": "Turquoise vs Orange",
                "clrA": "#10E38F",
                "clrB": "#FB7B08",
                "categoryName": "Ranked Modes",
                "colorsSwapped": false
            }
        }
    ]
}
