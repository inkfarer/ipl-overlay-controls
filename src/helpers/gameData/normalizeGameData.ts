import cloneDeep from 'lodash/cloneDeep';
import { GameData, RawGameData, StageModeList, StageNameList } from './gameData';
import { Locale } from '../../types/enums/Locale';

const commonStageNames: {[key in Locale]: { 'Unknown Stage': string, 'Counterpick': string } } = {
    [Locale.EN]: {
        'Unknown Stage': 'Unknown Stage',
        Counterpick: 'Counterpick'
    },
    [Locale.DE]: {
        'Unknown Stage': 'Unbekannte Arena',
        Counterpick: 'Counterpick'
    },
    [Locale.EU_FR]: {
        'Unknown Stage': 'Stage à déterminer',
        Counterpick: 'Counterpick'
    },
    [Locale.JA]: {
        'Unknown Stage': 'ステージ未選択',
        Counterpick: 'カウンターピック'
    }
};

const commonModeNames: {[key in Locale]: { 'Unknown Mode': string } } = {
    [Locale.EN]: {
        'Unknown Mode': 'Unknown Mode',
    },
    [Locale.DE]: {
        'Unknown Mode': 'Unbekannte Kampfart',
    },
    [Locale.EU_FR]: {
        'Unknown Mode': 'Mode à déterminer',
    },
    [Locale.JA]: {
        'Unknown Mode': 'ルール未選択'
    }
};

export function normalizeGameData<S extends readonly string[], M extends readonly string[]>(
    existingData: RawGameData<S, M>
): GameData<S, M> {
    const data = cloneDeep(existingData);

    return {
        ...data,
        stages: Object.entries(data.stages).reduce((result, [key, value]) => {
            result[key as Locale] = {
                ...value,
                ...commonStageNames[key as Locale]
            };

            return result;
        }, {} as StageNameList<S>),
        modes: Object.entries(data.modes).reduce((result, [key, value]) => {
            result[key as Locale] = {
                ...value,
                ...commonModeNames[key as Locale]
            };

            return result;
        }, {} as StageModeList<M>),
        colors: [
            ...data.colors.map(colorGroup => ({
                ...colorGroup,
                colors: colorGroup.colors.map(color => ({ ...color, isCustom: false }))
            })),
            {
                meta: {
                    name: 'Custom Color'
                },
                colors: [
                    {
                        index: 0,
                        title: 'Custom Color',
                        clrA: '#000000',
                        clrB: '#FFFFFF',
                        clrNeutral: '#818181',
                        isCustom: true
                    }
                ]
            }
        ]
    };
}
