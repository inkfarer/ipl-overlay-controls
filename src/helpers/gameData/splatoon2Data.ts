import { GameData } from './gameData';
import { normalizeGameData } from './normalizeGameData';
import { Locale } from '../../types/enums/Locale';

export const Splatoon2Stages = [
    'Ancho-V Games',
    'Arowana Mall',
    'Blackbelly Skatepark',
    'Camp Triggerfish',
    'Goby Arena',
    'Humpback Pump Track',
    'Inkblot Art Academy',
    'Kelp Dome',
    'MakoMart',
    'Manta Maria',
    'Moray Towers',
    'Musselforge Fitness',
    'New Albacore Hotel',
    'Piranha Pit',
    'Port Mackerel',
    'Shellendorf Institute',
    'Shifty Station',
    'Snapper Canal',
    'Starfish Mainstage',
    'Sturgeon Shipyard',
    'The Reef',
    'Wahoo World',
    'Walleye Warehouse',
    'Skipper Pavilion'
] as const;

export const Splatoon2Modes = [
    'Clam Blitz',
    'Tower Control',
    'Rainmaker',
    'Splat Zones',
    'Turf War'
] as const;

export const splatoon2Data: GameData<typeof Splatoon2Stages, typeof Splatoon2Modes> = normalizeGameData({
    stages: {
        [Locale.EN]: {
            'Ancho-V Games': 'Ancho-V Games',
            'Arowana Mall': 'Arowana Mall',
            'Blackbelly Skatepark': 'Blackbelly Skatepark',
            'Camp Triggerfish': 'Camp Triggerfish',
            'Goby Arena': 'Goby Arena',
            'Humpback Pump Track': 'Humpback Pump Track',
            'Inkblot Art Academy': 'Inkblot Art Academy',
            'Kelp Dome': 'Kelp Dome',
            MakoMart: 'MakoMart',
            'Manta Maria': 'Manta Maria',
            'Moray Towers': 'Moray Towers',
            'Musselforge Fitness': 'Musselforge Fitness',
            'New Albacore Hotel': 'New Albacore Hotel',
            'Piranha Pit': 'Piranha Pit',
            'Port Mackerel': 'Port Mackerel',
            'Shellendorf Institute': 'Shellendorf Institute',
            'Shifty Station': 'Shifty Station',
            'Snapper Canal': 'Snapper Canal',
            'Starfish Mainstage': 'Starfish Mainstage',
            'Sturgeon Shipyard': 'Sturgeon Shipyard',
            'The Reef': 'The Reef',
            'Wahoo World': 'Wahoo World',
            'Walleye Warehouse': 'Walleye Warehouse',
            'Skipper Pavilion': 'Skipper Pavilion'
        },
        [Locale.DE]: {
            'Ancho-V Games': 'Anchobit Games HQ',
            'Arowana Mall': 'Arowana Center',
            'Blackbelly Skatepark': 'Punkasius-Skatepark',
            'Camp Triggerfish': 'Camp Schützenfisch',
            'Goby Arena': 'Backfisch-Stadion',
            'Humpback Pump Track': 'Buckelwal-Piste',
            'Inkblot Art Academy': 'Perlmutt-Akademie',
            'Kelp Dome': 'Tümmlerkuppel',
            MakoMart: 'Cetacea-Markt',
            'Manta Maria': 'Manta Maria',
            'Moray Towers': 'Muränentürme',
            'Musselforge Fitness': 'Molluskelbude',
            'New Albacore Hotel': 'Hotel Neothun',
            'Piranha Pit': 'Steinköhler-Grube',
            'Port Mackerel': 'Heilbutt-Hafen',
            'Shellendorf Institute': 'Abyssal-Museum',
            'Shifty Station': 'Wandelzone',
            'Snapper Canal': 'Grätenkanal',
            'Starfish Mainstage': 'Seeigel-Rockbühne',
            'Sturgeon Shipyard': 'Störwerft',
            'The Reef': 'Korallenviertel',
            'Wahoo World': 'Flunder-Funpark',
            'Walleye Warehouse': 'Kofferfisch-Lager',
            'Skipper Pavilion': 'Grundel-Pavillon'
        },
        [Locale.EU_FR]: {
            'Ancho-V Games': 'Tentatec Studio',
            'Arowana Mall': 'Centre Arowana',
            'Blackbelly Skatepark': 'Skatepark Mako',
            'Camp Triggerfish': 'Hippo-Camping',
            'Goby Arena': 'Stade Bernique',
            'Humpback Pump Track': 'Piste Méroule',
            'Inkblot Art Academy': 'Institut Calam\'arts',
            'Kelp Dome': 'Serre Goémon',
            MakoMart: 'Supermarché Cétacé',
            'Manta Maria': 'Manta Maria',
            'Moray Towers': 'Tours Girelle',
            'Musselforge Fitness': 'Gymnase Ancrage',
            'New Albacore Hotel': 'Hôtel Atoll',
            'Piranha Pit': 'Carrière Caviar',
            'Port Mackerel': 'Docks Haddock',
            'Shellendorf Institute': 'Galerie des abysses',
            'Shifty Station': 'Plateforme polymorphe',
            'Snapper Canal': 'Canalamar',
            'Starfish Mainstage': 'Scène Sirène',
            'Sturgeon Shipyard': 'Chantier Narval',
            'The Reef': 'Allées salées',
            'Wahoo World': 'Parc Carapince',
            'Walleye Warehouse': 'Encrepôt',
            'Skipper Pavilion': 'Lagune aux gobies'
        },
        [Locale.JA]: {
            'Ancho-V Games': 'アンチョビットゲームズ',
            'Arowana Mall': 'アロワナモール',
            'Blackbelly Skatepark': 'Ｂバスパーク',
            'Camp Triggerfish': 'モンガラキャンプ場',
            'Goby Arena': 'アジフライスタジアム',
            'Humpback Pump Track': 'コンブトラック',
            'Inkblot Art Academy': '海女美術大学',
            'Kelp Dome': 'モズク農園',
            MakoMart: 'ザトウマーケット',
            'Manta Maria': 'マンタマリア号',
            'Moray Towers': 'タチウオパーキング',
            'Musselforge Fitness': 'フジツボスポーツクラブ',
            'New Albacore Hotel': 'ホテルニューオートロ',
            'Piranha Pit': 'ショッツル鉱山',
            'Port Mackerel': 'ホッケふ頭',
            'Shellendorf Institute': 'デボン海洋博物館',
            'Shifty Station': 'ミステリーゾーン',
            'Snapper Canal': 'エンガワ河川敷',
            'Starfish Mainstage': 'ガンガゼ野外音楽堂',
            'Sturgeon Shipyard': 'チョウザメ造船',
            'The Reef': 'バッテラストリート',
            'Wahoo World': 'スメーシーワールド',
            'Walleye Warehouse': 'ハコフグ倉庫',
            'Skipper Pavilion': 'ムツゴ楼'
        }
    },
    stageImagePaths: {
        'Ancho-V Games': 'splatoon2/S2_Stage_Ancho-V_Games.webp',
        'Arowana Mall': 'splatoon2/S2_Stage_Arowana_Mall.webp',
        'Blackbelly Skatepark': 'splatoon2/S2_Stage_Blackbelly_Skatepark.webp',
        'Camp Triggerfish': 'splatoon2/S2_Stage_Camp_Triggerfish.webp',
        'Goby Arena': 'splatoon2/S2_Stage_Goby_Arena.webp',
        'Humpback Pump Track': 'splatoon2/S2_Stage_Humpback_Pump_Track.webp',
        'Inkblot Art Academy': 'splatoon2/S2_Stage_Inkblot_Art_Academy.webp',
        'Kelp Dome': 'splatoon2/S2_Stage_Kelp_Dome.webp',
        MakoMart: 'splatoon2/S2_Stage_MakoMart.webp',
        'Manta Maria': 'splatoon2/S2_Stage_Manta_Maria.webp',
        'Moray Towers': 'splatoon2/S2_Stage_Moray_Towers.webp',
        'Musselforge Fitness': 'splatoon2/S2_Stage_Musselforge_Fitness.webp',
        'New Albacore Hotel': 'splatoon2/S2_Stage_New_Albacore_Hotel.webp',
        'Piranha Pit': 'splatoon2/S2_Stage_Piranha_Pit.webp',
        'Port Mackerel': 'splatoon2/S2_Stage_Port_Mackerel.webp',
        'Shellendorf Institute': 'splatoon2/S2_Stage_Shellendorf_Institute.webp',
        'Shifty Station': 'splatoon2/S2_Stage_Shifty_Station.webp',
        'Snapper Canal': 'splatoon2/S2_Stage_Snapper_Canal.webp',
        'Starfish Mainstage': 'splatoon2/S2_Stage_Starfish_Mainstage.webp',
        'Sturgeon Shipyard': 'splatoon2/S2_Stage_Sturgeon_Shipyard.webp',
        'The Reef': 'splatoon2/S2_Stage_The_Reef.webp',
        'Wahoo World': 'splatoon2/S2_Stage_Wahoo_World.webp',
        'Walleye Warehouse': 'splatoon2/S2_Stage_Walleye_Warehouse.webp',
        'Skipper Pavilion': 'splatoon2/S2_Stage_Skipper_Pavilion.webp',
    },
    modes: {
        [Locale.EN]: {
            'Clam Blitz': 'Clam Blitz',
            'Tower Control': 'Tower Control',
            Rainmaker: 'Rainmaker',
            'Splat Zones': 'Splat Zones',
            'Turf War': 'Turf War'
        },
        [Locale.DE]: {
            'Clam Blitz': 'Muschelchaos',
            'Tower Control': 'Turmkommando',
            Rainmaker: 'Operation Goldfisch',
            'Splat Zones': 'Herrschaft',
            'Turf War': 'Revierkampf'
        },
        [Locale.EU_FR]: {
            'Clam Blitz': 'Pluie de palourdes',
            'Tower Control': 'Expédition risquée',
            Rainmaker: 'Mission Bazookarpe',
            'Splat Zones': 'Défense de zone',
            'Turf War': 'Guerre de territoire'
        },
        [Locale.JA]: {
            'Clam Blitz': 'ガチアサリ',
            'Tower Control': 'ガチヤグラ',
            Rainmaker: 'ガチホコバトル',
            'Splat Zones': 'ガチエリア',
            'Turf War': 'ナワバリバトル'
        }
    },
    colors: [
        {
            meta: {
                name: 'Ranked Modes',
                key: 'rankedModes'
            },
            colors: [
                {
                    index: 0,
                    title: 'Green vs Grape',
                    key: 'greenGrape',
                    clrA: '#37FC00',
                    clrB: '#7D28FC',
                    clrNeutral: '#F4067E'
                },
                {
                    index: 1,
                    title: 'Green vs Magenta',
                    key: 'greenMagenta',
                    clrA: '#04D976',
                    clrB: '#D600AB',
                    clrNeutral: '#D2E500'
                },
                {
                    index: 2,
                    title: 'Turquoise vs Orange',
                    key: 'turquoiseOrange',
                    clrA: '#10E38F',
                    clrB: '#FB7B08',
                    clrNeutral: '#6912CD'
                },
                {
                    index: 3,
                    title: 'Mustard vs Purple',
                    key: 'mustardPurple',
                    clrA: '#FF9E03',
                    clrB: '#B909E0',
                    clrNeutral: '#08C66B'
                },
                {
                    index: 4,
                    title: 'Dark Blue vs Green',
                    key: 'darkBlueGreen',
                    clrA: '#2F27CC',
                    clrB: '#37FC00',
                    clrNeutral: '#EA01B7'
                },
                {
                    index: 5,
                    title: 'Purple vs Green',
                    key: 'purpleGreen',
                    clrA: '#B909E0',
                    clrB: '#37FC00',
                    clrNeutral: '#F87604'
                },
                {
                    index: 6,
                    title: 'Yellow vs Blue',
                    key: 'yellowBlue',
                    clrA: '#FEF232',
                    clrB: '#2ED2FE',
                    clrNeutral: '#FD5600'
                }
            ]
        },
        {
            meta: {
                name: 'Turf War',
                key: 'turfWar'
            },
            colors: [
                {
                    index: 0,
                    title: 'Yellow vs Purple',
                    key: 'yellowPurple',
                    clrA: '#D1E004',
                    clrB: '#960CB2',
                    clrNeutral: '#0EB962'
                },
                {
                    index: 1,
                    title: 'Pink vs Blue',
                    key: 'pinkBlue',
                    clrA: '#E61077',
                    clrB: '#361CB8',
                    clrNeutral: '#24C133'
                },
                {
                    index: 2,
                    title: 'Pink vs Yellow',
                    key: 'pinkYellow',
                    clrA: '#ED0C6A',
                    clrB: '#D5E802',
                    clrNeutral: '#08C24D'
                },
                {
                    index: 3,
                    title: 'Purple vs Turquoise',
                    key: 'purpleTurquoise',
                    clrA: '#6B10CC',
                    clrB: '#08CC81',
                    clrNeutral: '#EB246D'
                },
                {
                    index: 4,
                    title: 'Pink vs Light Blue',
                    key: 'pinkLightBlue',
                    clrA: '#E30960',
                    clrB: '#02ADCF',
                    clrNeutral: '#DDE713'
                },
                {
                    index: 5,
                    title: 'Purple vs Orange',
                    key: 'purpleOrange',
                    clrA: '#5617C2',
                    clrB: '#FF5F03',
                    clrNeutral: '#ACE81E'
                },
                {
                    index: 6,
                    title: 'Pink vs Green',
                    key: 'pinkGreen',
                    clrA: '#E60572',
                    clrB: '#1BBF0F',
                    clrNeutral: '#CCE50C'
                },
                {
                    index: 7,
                    title: 'Yellow vs Blue',
                    key: 'yellowBlue',
                    clrA: '#F1CE33',
                    clrB: '#4B12BE',
                    clrNeutral: '#E62E96'
                }
            ]
        },
        {
            meta: {
                name: 'Color Lock',
                key: 'colorLock'
            },
            colors: [
                {
                    index: 0,
                    title: 'Yellow vs Blue (Color Lock)',
                    key: 'yellowBlue',
                    clrA: '#FEF232',
                    clrB: '#2F27CC',
                    clrNeutral: '#DC1589'
                }
            ]
        }
    ]
});
