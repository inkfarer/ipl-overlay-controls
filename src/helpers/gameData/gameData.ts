import { ColorGroup } from '../../types/colors';
import { GameVersion } from '../../types/enums/gameVersion';
import { splatoon2Data, Splatoon2Modes, Splatoon2Stages } from './splatoon2Data';
import { splatoon3Data, Splatoon3Modes, Splatoon3Stages } from './splatoon3Data';
import { Locale } from '../../types/enums/Locale';

export interface RawGameData<S, M> {
    stages: {[locale in Locale]: {[stage in keyof S]: string } },
    stageImagePaths: {[stage in keyof S]: string },
    modes: {[locale in Locale]: {[mode in keyof M]: string } },
    colors: ColorGroup[]
}

export type StageNameList<S> = {
    [locale in Locale]: {
        [stage in keyof S]: string & { 'Unknown Stage': string, Counterpick: string }
    }
};

export type StageModeList<M> = {
    [locale in Locale]: {
        [mode in keyof M]: string & { 'Unknown Mode': string }
    }
};

export interface GameData<S, M> {
    stages: StageNameList<S>,
    stageImagePaths: {[stage in keyof S]: string },
    modes: StageModeList<M>,
    colors: ColorGroup[]
}

type StageTypeMap = { [GameVersion.SPLATOON_2]: Splatoon2Stages, [GameVersion.SPLATOON_3]: Splatoon3Stages };
type ModeTypeMap = { [GameVersion.SPLATOON_2]: Splatoon2Modes, [GameVersion.SPLATOON_3]: Splatoon3Modes };

export const perGameData: {[key in GameVersion]: GameData<StageTypeMap[key], ModeTypeMap[key]>} = {
    [GameVersion.SPLATOON_2]: splatoon2Data,
    [GameVersion.SPLATOON_3]: splatoon3Data
};
