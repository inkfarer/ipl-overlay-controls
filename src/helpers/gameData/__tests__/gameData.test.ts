import { perGameData } from '../gameData';

describe('gameData', () => {
    describe('perGameData', () => {
        it('matches snapshot', () => {
            expect(perGameData).toMatchSnapshot();
        });
    });
});
