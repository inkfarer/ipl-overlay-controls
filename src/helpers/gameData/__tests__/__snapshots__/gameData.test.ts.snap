// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gameData perGameData matches snapshot 1`] = `
Object {
  "SPLATOON_2": Object {
    "colors": Array [
      Object {
        "colors": Array [
          Object {
            "clrA": "#37FC00",
            "clrB": "#7D28FC",
            "clrNeutral": "#F4067E",
            "index": 0,
            "isCustom": false,
            "title": "Green vs Grape",
          },
          Object {
            "clrA": "#04D976",
            "clrB": "#D600AB",
            "clrNeutral": "#D2E500",
            "index": 1,
            "isCustom": false,
            "title": "Green vs Magenta",
          },
          Object {
            "clrA": "#10E38F",
            "clrB": "#FB7B08",
            "clrNeutral": "#6912CD",
            "index": 2,
            "isCustom": false,
            "title": "Turquoise vs Orange",
          },
          Object {
            "clrA": "#FF9E03",
            "clrB": "#B909E0",
            "clrNeutral": "#08C66B",
            "index": 3,
            "isCustom": false,
            "title": "Mustard vs Purple",
          },
          Object {
            "clrA": "#2F27CC",
            "clrB": "#37FC00",
            "clrNeutral": "#EA01B7",
            "index": 4,
            "isCustom": false,
            "title": "Dark Blue vs Green",
          },
          Object {
            "clrA": "#B909E0",
            "clrB": "#37FC00",
            "clrNeutral": "#F87604",
            "index": 5,
            "isCustom": false,
            "title": "Purple vs Green",
          },
          Object {
            "clrA": "#FEF232",
            "clrB": "#2ED2FE",
            "clrNeutral": "#FD5600",
            "index": 6,
            "isCustom": false,
            "title": "Yellow vs Blue",
          },
        ],
        "meta": Object {
          "name": "Ranked Modes",
        },
      },
      Object {
        "colors": Array [
          Object {
            "clrA": "#D1E004",
            "clrB": "#960CB2",
            "clrNeutral": "#0EB962",
            "index": 0,
            "isCustom": false,
            "title": "Yellow vs Purple",
          },
          Object {
            "clrA": "#E61077",
            "clrB": "#361CB8",
            "clrNeutral": "#24C133",
            "index": 1,
            "isCustom": false,
            "title": "Pink vs Blue",
          },
          Object {
            "clrA": "#ED0C6A",
            "clrB": "#D5E802",
            "clrNeutral": "#08C24D",
            "index": 2,
            "isCustom": false,
            "title": "Pink vs Yellow",
          },
          Object {
            "clrA": "#6B10CC",
            "clrB": "#08CC81",
            "clrNeutral": "#EB246D",
            "index": 3,
            "isCustom": false,
            "title": "Purple vs Turquoise",
          },
          Object {
            "clrA": "#E30960",
            "clrB": "#02ADCF",
            "clrNeutral": "#DDE713",
            "index": 4,
            "isCustom": false,
            "title": "Pink vs Light Blue",
          },
          Object {
            "clrA": "#5617C2",
            "clrB": "#FF5F03",
            "clrNeutral": "#ACE81E",
            "index": 5,
            "isCustom": false,
            "title": "Purple vs Orange",
          },
          Object {
            "clrA": "#E60572",
            "clrB": "#1BBF0F",
            "clrNeutral": "#CCE50C",
            "index": 6,
            "isCustom": false,
            "title": "Pink vs Green",
          },
          Object {
            "clrA": "#F1CE33",
            "clrB": "#4B12BE",
            "clrNeutral": "#E62E96",
            "index": 7,
            "isCustom": false,
            "title": "Yellow vs Blue",
          },
        ],
        "meta": Object {
          "name": "Turf War",
        },
      },
      Object {
        "colors": Array [
          Object {
            "clrA": "#FEF232",
            "clrB": "#2F27CC",
            "clrNeutral": "#DC1589",
            "index": 0,
            "isCustom": false,
            "title": "Yellow vs Blue (Color Lock)",
          },
        ],
        "meta": Object {
          "name": "Color Lock",
        },
      },
      Object {
        "colors": Array [
          Object {
            "clrA": "#000000",
            "clrB": "#FFFFFF",
            "clrNeutral": "#818181",
            "index": 0,
            "isCustom": true,
            "title": "Custom Color",
          },
        ],
        "meta": Object {
          "name": "Custom Color",
        },
      },
    ],
    "modes": Object {
      "DE": Object {
        "Clam Blitz": "Muschelchaos",
        "Rainmaker": "Operation Goldfisch",
        "Splat Zones": "Herrschaft",
        "Tower Control": "Turmkommando",
        "Turf War": "Revierkampf",
        "Unknown Mode": "Unbekannte Kampfart",
      },
      "EN": Object {
        "Clam Blitz": "Clam Blitz",
        "Rainmaker": "Rainmaker",
        "Splat Zones": "Splat Zones",
        "Tower Control": "Tower Control",
        "Turf War": "Turf War",
        "Unknown Mode": "Unknown Mode",
      },
    },
    "stages": Object {
      "DE": Object {
        "Ancho-V Games": "Anchobit Games HQ",
        "Arowana Mall": "Arowana Center",
        "Blackbelly Skatepark": "Punkasius-Skatepark",
        "Camp Triggerfish": "Camp Schützenfisch",
        "Counterpick": "Counterpick",
        "Goby Arena": "Backfisch-Stadion",
        "Humpback Pump Track": "Buckelwal-Piste",
        "Inkblot Art Academy": "Perlmutt-Akademie",
        "Kelp Dome": "Tümmlerkuppel",
        "MakoMart": "Cetacea-Markt",
        "Manta Maria": "Manta Maria",
        "Moray Towers": "Muränentürme",
        "Musselforge Fitness": "Molluskelbude",
        "New Albacore Hotel": "Hotel Neothun",
        "Piranha Pit": "Steinköhler-Grube",
        "Port Mackerel": "Heilbutt-Hafen",
        "Shellendorf Institute": "Abyssal-Museum",
        "Shifty Station": "Wandelzone",
        "Skipper Pavilion": "Grundel-Pavillon",
        "Snapper Canal": "Grätenkanal",
        "Starfish Mainstage": "Seeigel-Rockbühne",
        "Sturgeon Shipyard": "Störwerft",
        "The Reef": "Korallenviertel",
        "Unknown Stage": "Unbekannte Arena",
        "Wahoo World": "Flunder-Funpark",
        "Walleye Warehouse": "Kofferfisch-Lager",
      },
      "EN": Object {
        "Ancho-V Games": "Ancho-V Games",
        "Arowana Mall": "Arowana Mall",
        "Blackbelly Skatepark": "Blackbelly Skatepark",
        "Camp Triggerfish": "Camp Triggerfish",
        "Counterpick": "Counterpick",
        "Goby Arena": "Goby Arena",
        "Humpback Pump Track": "Humpback Pump Track",
        "Inkblot Art Academy": "Inkblot Art Academy",
        "Kelp Dome": "Kelp Dome",
        "MakoMart": "MakoMart",
        "Manta Maria": "Manta Maria",
        "Moray Towers": "Moray Towers",
        "Musselforge Fitness": "Musselforge Fitness",
        "New Albacore Hotel": "New Albacore Hotel",
        "Piranha Pit": "Piranha Pit",
        "Port Mackerel": "Port Mackerel",
        "Shellendorf Institute": "Shellendorf Institute",
        "Shifty Station": "Shifty Station",
        "Skipper Pavilion": "Skipper Pavilion",
        "Snapper Canal": "Snapper Canal",
        "Starfish Mainstage": "Starfish Mainstage",
        "Sturgeon Shipyard": "Sturgeon Shipyard",
        "The Reef": "The Reef",
        "Unknown Stage": "Unknown Stage",
        "Wahoo World": "Wahoo World",
        "Walleye Warehouse": "Walleye Warehouse",
      },
    },
  },
  "SPLATOON_3": Object {
    "colors": Array [
      Object {
        "colors": Array [
          Object {
            "clrA": "#F753A1",
            "clrB": "#35CC2D",
            "clrNeutral": "#4A36FF",
            "index": 0,
            "isCustom": false,
            "title": "Pink vs Green",
          },
          Object {
            "clrA": "#EBCC31",
            "clrB": "#A032DB",
            "clrNeutral": "#73DE35",
            "index": 1,
            "isCustom": false,
            "title": "Gold vs Purple",
          },
          Object {
            "clrA": "#2ADBC3",
            "clrB": "#E34984",
            "clrNeutral": "#7577FF",
            "index": 2,
            "isCustom": false,
            "title": "Turquoise vs Pink",
          },
          Object {
            "clrA": "#F56522",
            "clrB": "#821CD6",
            "clrNeutral": "#CDFF45",
            "index": 3,
            "isCustom": false,
            "title": "Orange vs Purple",
          },
          Object {
            "clrA": "#EEFC58",
            "clrB": "#7635F0",
            "clrNeutral": "#54FDE8",
            "index": 4,
            "isCustom": false,
            "title": "Yellow vs Purple",
          },
          Object {
            "clrA": "#FC7735",
            "clrB": "#4048DB",
            "clrNeutral": "#F8F755",
            "index": 5,
            "isCustom": false,
            "title": "Orange vs Dark Blue",
          },
          Object {
            "clrA": "#2ADBC6",
            "clrB": "#FA5A41",
            "clrNeutral": "#4445FF",
            "index": 6,
            "isCustom": false,
            "title": "Turquoise vs Orange",
          },
          Object {
            "clrA": "#EDD921",
            "clrB": "#461EE6",
            "clrNeutral": "#4445FF",
            "index": 7,
            "isCustom": false,
            "title": "Yellow vs Dark Blue",
          },
          Object {
            "clrA": "#2C2CDB",
            "clrB": "#F29C33",
            "clrNeutral": "#FF67EE",
            "index": 8,
            "isCustom": false,
            "title": "Dark Blue vs Orange",
          },
          Object {
            "clrA": "#B3DE45",
            "clrB": "#D43BCA",
            "clrNeutral": "#FFC042",
            "index": 9,
            "isCustom": false,
            "title": "Green vs Pink",
          },
        ],
        "meta": Object {
          "name": "Ranked Modes",
        },
      },
      Object {
        "colors": Array [
          Object {
            "clrA": "#000000",
            "clrB": "#FFFFFF",
            "clrNeutral": "#818181",
            "index": 0,
            "isCustom": true,
            "title": "Custom Color",
          },
        ],
        "meta": Object {
          "name": "Custom Color",
        },
      },
    ],
    "modes": Object {
      "DE": Object {
        "Clam Blitz": "Muschelchaos",
        "Rainmaker": "Operation Goldfisch",
        "Splat Zones": "Herrschaft",
        "Tower Control": "Turmkommando",
        "Turf War": "Revierkampf",
        "Unknown Mode": "Unbekannte Kampfart",
      },
      "EN": Object {
        "Clam Blitz": "Clam Blitz",
        "Rainmaker": "Rainmaker",
        "Splat Zones": "Splat Zones",
        "Tower Control": "Tower Control",
        "Turf War": "Turf War",
        "Unknown Mode": "Unknown Mode",
      },
    },
    "stages": Object {
      "DE": Object {
        "Counterpick": "Counterpick",
        "Eeltail Alley": "Streifenaal-Straße",
        "Hagglefish Market": "Schnapperchen-Basar",
        "Hammerhead Bridge": "Makrelenbrücke",
        "Inkblot Art Academy": "Perlmutt-Akademie",
        "Mahi-Mahi Resort": "Mahi-Mahi-Resort",
        "MakoMart": "Cetacea-Markt",
        "Mincemeat Metalworks": "Aalstahl-Metallwerk",
        "Museum d'Alfonsino": "Pinakoithek",
        "Scorch Gorge": "Sengkluft",
        "Sturgeon Shipyard": "Störwerft",
        "Undertow Spillway": "Schwertmuschel-Reservoir",
        "Unknown Stage": "Unbekannte Arena",
        "Wahoo World": "Flunder-Funpark",
      },
      "EN": Object {
        "Counterpick": "Counterpick",
        "Eeltail Alley": "Eeltail Alley",
        "Hagglefish Market": "Hagglefish Market",
        "Hammerhead Bridge": "Hammerhead Bridge",
        "Inkblot Art Academy": "Inkblot Art Academy",
        "Mahi-Mahi Resort": "Mahi-Mahi Resort",
        "MakoMart": "MakoMart",
        "Mincemeat Metalworks": "Mincemeat Metalworks",
        "Museum d'Alfonsino": "Museum d'Alfonsino",
        "Scorch Gorge": "Scorch Gorge",
        "Sturgeon Shipyard": "Sturgeon Shipyard",
        "Undertow Spillway": "Undertow Spillway",
        "Unknown Stage": "Unknown Stage",
        "Wahoo World": "Wahoo World",
      },
    },
  },
}
`;
